openedx/core/djangoapps/extensions_lite/management/commands/reset_db.py

```diff
@@ -0,0 +1,190 @@
+            connection.set_isolation_level(0)  # autocommit false
+            cursor = connection.cursor()
+
+            drop_query = "DROP DATABASE \"%s\";" % database_name
+            logging.info('Executing... "' + drop_query + '"')
+            try:
+                cursor.execute(drop_query)
+            except Database.ProgrammingError as e:
+                logging.exception("Error: %s" % str(e))
+
+            create_query = "CREATE DATABASE \"%s\"" % database_name
+            if owner:
+                create_query += " WITH OWNER = \"%s\" " % owner
+            create_query += " ENCODING = 'UTF8'"
+
+            if engine == 'postgis' and django.VERSION < (1, 9):
+                # For PostGIS 1.5, fetch template name if it exists
+                from django.contrib.gis.db.backends.postgis.base import DatabaseWrapper
+                postgis_template = DatabaseWrapper(dbinfo).template_postgis
+                if postgis_template is not None:
+                    create_query += ' TEMPLATE = %s' % postgis_template
+
+            if settings.DEFAULT_TABLESPACE:
+                create_query += ' TABLESPACE = %s;' % settings.DEFAULT_TABLESPACE
+            else:
+                create_query += ';'
+
+            logging.info('Executing... "' + create_query + '"')
+            cursor.execute(create_query)
+
+        else:
+            raise CommandError("Unknown database engine %s" % engine)
+
+        if verbosity >= 2:
+            print("Reset successful.")
+
+
+def parse_mysql_cnf(dbinfo):
+    """
+    Attempt to parse mysql database config file for connection settings.
+    Ideally we would hook into django's code to do this, but read_default_file is handled by the mysql C libs
+    so we have to emulate the behaviour
+
+    Settings that are missing will return ''
+    returns (user, password, database_name, database_host, database_port)
+    """
+    read_default_file = dbinfo.get('OPTIONS', {}).get('read_default_file')
+    if read_default_file:
+        config = configparser.RawConfigParser({
+            'user': '',
+            'password': '',
+            'database': '',
+            'host': '',
+            'port': '',
+            'socket': '',
+        })
+        import os
+        config.read(os.path.expanduser(read_default_file))
+        try:
+            user = config.get('client', 'user')
+            password = config.get('client', 'password')
+            database_name = config.get('client', 'database')
+            database_host = config.get('client', 'host')
+            database_port = config.get('client', 'port')
+            socket = config.get('client', 'socket')
+
+            if database_host == 'localhost' and socket:
+                # mysql actually uses a socket if host is localhost
+                database_host = socket
+
+            return user, password, database_name, database_host, database_port
+
+        except configparser.NoSectionError:
+            pass
```

Comment:
I would expect that if the conf is included in settings it should be usable here. Maybe log it if things fall through here?